src/psaform/psaform.jsx

```diff
@@ -0,0 +1,245 @@
+
+  useEffect(() => {
+    // Check for required fields whenever formData changes
+    const { state } = checkDisabled();
+    setIsSubmitDisabled(state);
+  }, [formData]);
+
+  useEffect(() => {
+    pirschAnalytics("Visited Page", { meta: { visited: "Feedback" } });
+  }, [consentRedux]);
+
+  const convertMessageArr = (arr) => {
+    if (arr.length === 0) return "";
+    if (arr.length === 1) return `The "${arr[0]}" field is blank`;
+    if (arr.length === 2) return `The "${arr.join('" and "')}" fields are blank`;
+    return `The "${arr.slice(0, -1).join('", "')}", and "${arr[arr.length - 1]}" fields are blank`;
+  };
+
+  const checkDisabled = () => {
+    const messageArr = [];
+  
+    // Check required text fields
+    textFields.forEach((field) => {
+      if (field.required && formData[field.name] === "") {
+        messageArr.push(field.label);
+      }
+    });
+  
+    // Check required checkboxes in each group
+    checkboxes.forEach((group) => {
+      if(group.required && formData.labels.length === 0)
+      {
+        messageArr.push(group.title);
+      }
+    });
+  
+    const messageStr = convertMessageArr(messageArr);
+    return messageArr.length > 0
+      ? { state: true, message: messageStr }
+      : { state: false, message: "" };
+  }; 
+
+  const handleTextInputChange = (event, name) => {
+    setFormData({ ...formData, [name]: event.target.value });
+  };
+
+  const handleCheckboxChange = (event) => {
+    const { name, checked } = event.target;
+    setFormData((prev) => ({
+      ...prev,
+      labels: checked ? [...(prev.labels || []), name] : remove(prev.labels || [], name),
+    }));
+  };
+
+  const handleSubmit = () => {
+    const { state, message } = checkDisabled();
+    if (state) {
+      setSnackbarData({ open: true, message, color: "red" });
+      return;
+    }
+
+    fetch(apiUrl, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(formData),
+    })
+      .then((response) => {
+        setSnackbarData({
+          open: true,
+          message: response.status === 201
+            ? submitMessage
+            : `Error ${response.status}. ${
+                response.status === 400
+                  ? "Bad Request"
+                  : response.status === 422
+                  ? "Unprocessable Entry"
+                  : "Internal Server Error"
+              }`,
+          color: response.status === 201 ? "green" : "red",
+        });
+        return response.json();
+      })
+      .catch((error) => console.error(error));
+  };
+
+  return (
+    <Grid container rowSpacing={5} style={{ padding: "3% 10%", textAlign: "left" }}>
+      <Grid container item spacing={1} justifyContent="center">
+        <Grid item xs={12}>
+          <Typography variant="h3">{headerTitle}</Typography>
+        </Grid>
+      </Grid>
+
+      {/* Dynamically Render Text Fields */}
+      {textFields.map((field, index) => (
```

Comment:
How things are currently made we have textFields and Checkboxes (line 155) separate.  This means that if we have a form that needs a text field then a checkbox then a textfield we will not be able to create that.  

What i recommend is create a single prop called formContent and that should be an array of objects.  One of the properties of each object should be type.  This will also not limit us to text boxes and checkboxes, we can then create an endless amount of types as new forms are created.  Then we just have to add logic to handle the new types.  

let me know if you'd like a little more guidance in getting this done.